examples/apps/qr_server.py

```
@@ -0,0 +1,164 @@
+    ui=ResourceUI(csp=ResourceCSP(resource_domains=["https://unpkg.com"])),
+)
+def view() -> str:
+    """Interactive QR code viewer — renders tool results as images."""
+    return EMBEDDED_VIEW_HTML
+
+
+if __name__ == "__main__":
+    if "--stdio" in sys.argv:
+        mcp.run(transport="stdio")
+    else:
+        mcp.run(transport="streamable-http", host="0.0.0.0", port=3001)
```

Ruff (0.14.14)

[error] Possible binding to all interfaces

(S104)